WordPress/openverse-frontend
# Prevent hiding and reappearing of 'Load more' button and meta search

## Fixes


Fixes #1539 by @dhruvkb (identified by @obulat)

## Description



The scroll shift described in the issue occurs because the 'Load more button' and the meta search box at the bottom of the page disappear during the loading and then reappear. This can be fixed by turning off the conditional hiding and showing the button and the meta search box at all times.

This PR:
- keeps the meta search buttons visible when loading more results.
- keeps the loading button visible (but disabled) when loading more results. The text is changed to 'Loading...' and it doubles as an activity indicator.
- hides the loading button when it has been established that there are no more results to load.

## Testing Instructions


Search for a popular term with lots of results like 'bird'. Then scroll to the bottom and hit 'Load more results'. You will see that the aforementioned elements remain visible and the jumpy scroll is not present.

Search for a less popular term with less than 1 page of results, like 'birdy boo'. The 'Load more results' button will be missing.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
